ios/FluentUI/Bottom Sheet/BottomSheetController.swift

```diff
@@ -379,8 +391,101 @@ public class BottomSheetController: UIViewController {
         constraints.append(contentsOf: makeLayoutGuideConstraints())
 
         NSLayoutConstraint.activate(constraints)
+
+        updateAppearance()
+    }
+
+    public override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        tokenSet.update(fluentTheme)
+    }
+
+    public override func viewDidLayoutSubviews() {
+        let newHeight = view.bounds.height
+        if currentRootViewHeight != newHeight && currentExpansionState == .transitioning {
+            // The view height has changed and we can't guarantee the animation target frame is valid anymore.
+            // Let's complete animations and cancel ongoing gestures to guarantee we end up in a good state.
+            completeAnimationsIfNeeded(skipToEnd: true)
+
+            if panGestureRecognizer.state != .possible {
+                panGestureRecognizer.state = .cancelled
+            }
+        }
+        currentRootViewHeight = newHeight
+
+        // In the transitioning state a pan gesture or an animator temporarily owns the sheet frame updates,
+        // so to avoid interfering we won't update the frame here.
+        if currentExpansionState != .transitioning {
+            bottomSheetView.frame = sheetFrame(offset: offset(for: currentExpansionState))
+            updateSheetLayoutGuideTopConstraint()
+            updateExpandedContentAlpha()
+            updateDimmingViewAlpha()
+            updateDimmingViewAccessibility()
+        }
+        collapsedHeightInSafeArea = view.safeAreaLayoutGuide.layoutFrame.maxY - offset(for: .collapsed)
+
+        super.viewDidLayoutSubviews()
+    }
+
+    public override func viewSafeAreaInsetsDidChange() {
+        super.viewSafeAreaInsetsDidChange()
+        completeAnimationsIfNeeded(skipToEnd: true)
+    }
+
+    public override func willTransition(to newCollection: UITraitCollection, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.willTransition(to: newCollection, with: coordinator)
+        completeAnimationsIfNeeded(skipToEnd: true)
+    }
+
+    public typealias TokenSetKeyType = BottomSheetTokenSet.Tokens
+    public var tokenSet: BottomSheetTokenSet = .init()
+
+    var tokenSetSink: AnyCancellable?
+    var fluentTheme: FluentTheme { return view.fluentTheme }
+
+    private func updateAppearance() {
+        updateBackgroundColor()
+        updateShadows()
+        updateCornerRadius()
+    }
+
+    private func updateBackgroundColor() {
+        let backgroundColor = UIColor(dynamicColor: tokenSet[.backgroundColor].dynamicColor)
+        bottomSheetView.subviews[0].backgroundColor = backgroundColor
+        overflowView.backgroundColor = backgroundColor
     }
 
+    private func updateShadows() {
+        let shadowInfo = tokenSet[.shadow].shadowInfo
+        if let ambientShadow = bottomSheetView.layer.sublayers?[1] {
```

Comment:
If we do have sublayers (not nil), is it guaranteed that there will always be 2 items in the layer?